backend/src/main/java/com/capstone/moneytree/service/impl/DefaultUserService.java

```diff
@@ -374,6 +380,32 @@ public void deleteUserByEmail(String email) {
         userDao.delete(existingUser);
     }
 
+    @Override
+    public List<User> getTopUsers(String symbol) {
+        List<User> users = new ArrayList<>();
+        // get a list of all user who own a stock with this symbol
+        ownsDao.findAll().stream()
+                .filter(owns -> owns.getStock().getSymbol().equals(symbol))
+                .collect(toList())
+                .forEach(owns -> users.add(owns.getUser()));
+
+        if (users.isEmpty()) {
+            return emptyList();
+        }
+
+        // sort user by ascending score order
+        Collections.sort(users);
+
+        // reverse to get biggest score first
+        Collections.reverse(users);
+
+        int top10percent = users.size() / 10; // rounded down for conservative results
+
+        return users.stream()
+                .limit(top10percent)
+                .collect(toList());
+    }
+
```

Comment:
For now, we can afford to do the query on server side since the number of users/relationship is low and not a bottleneck. The moment we hit more users, we would need to transfer this query to database layer

Reply:
I also floor down the top 10% for conservative values so if we have 95 users, then the top 10% is 9 users

Reply:
Yeah this will be server heavy. Let`s see how many users we get.